evidence.go

```diff
@@ -53,3 +57,34 @@ func (e *Evidence) AddProcess(p Process) error {
 
 	return nil
 }
+
+// Valid validates the Evidence receiver to ensure it has valid required fields
+func (e Evidence) Valid() error {
+	if e.DeviceID == "" {
+		return errors.New("evidence device-id is empty")
+	}
+
+	if e.Date.IsZero() {
+		return errors.New("evidence date is zero")
+	}
+
+	// Validate Files if present
```

Comment:
The correct implementation is to invoke Method Valid() on the File Object and let File Object, in the file file.go to implement the valid file that can be invoked from this place. In the Valid for file, check for validity of  Mandatory and check for presence of optional elements in the file object. If Optional Elements are present, then please check their validity!